script.js

```diff
@@ -1,4 +1,368 @@
+ * @returns {Array} anArray sorted in ascending order
+ */
+var sortAnArray = function (anArray) {
+  // get the length of the given array
+  var length = anArray.length;
+
+  //Use bubble sort to sort the array
+  for (var i = 0; i < length - 1; i += 1) {
+    for (var j = 0; j < length - 1; j += 1) {
+      if (anArray[j] > anArray[j + 1]) {
+        var temp = anArray[j];
+        anArray[j] = anArray[j + 1];
+        anArray[j + 1] = temp;
+      }
+    }
+  }
+  return anArray;
+};
+
+/**
+ * Automate choosing the player's number from the dice rolls
+ * Return the player number
+ * @param {number} firstNumeralIndex
+ */
+var getPlayerNumberAutomatically = function () {
+  // Get the current player's dice array.
+  // We use the ternary operator on the following line for more concise syntax.
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator
+  // The example below is equivalent to:
+  // var diceArray;
+  // if (currPlayer === 1) { diceArray = player1Dice; } else { diceArray = player2Dice; }
+  var diceArray = currPlayer === 1 ? player1Dice : player2Dice;
+  var playerNum;
+  // // Sort the array such that it can order any number of elements in ascending order
+  sortAnArray(diceArray);
+
+  // By now, dice array would be sorted. Iterate thru diceArray again, this time concatenating the elements from largest to smallest
+  while (diceArray.length > 0) {
+    // get hold of the largest num in the array (which is also the last element since we sorted it in ascending order earlier)
+    var largestNum = diceArray.pop();
+    // On the first iteration, there is no need to concatenate; just reassign playerNum
+    if (playerNum == null) {
+      playerNum = largestNum;
+    }
+    // else if it's not the first iteration, concatenate to get the lowest possible number
+    else {
+      // if the mode is LOWEST_COMBINED_NUMBER, create the lowest number
+      if (mode == LOWEST_COMBINED_NUMBER) {
+        playerNum = concatenate2Numbers(largestNum, playerNum);
+      }
+      // else it must be REGULAR mode, so concatenate to get the largest possible number
+      else {
+        playerNum = concatenate2Numbers(playerNum, largestNum);
+      }
+    }
+  }
+  // Update player1Num or player2Num to reflect the current round's score
+  if (currPlayer === 1) {
+    player1Num = playerNum;
+  } else {
+    player2Num = playerNum;
+  }
+  // Return generated player num to parent function
+  return playerNum;
+};
+
+/**
+ * Update the global variable that tracks  players' current round number
+ * @param {number} firstNumeralIndex
+ */
+updateCurrentRoundRoll = function (playerNum) {
+  // Store player num in the relevant global player num variable
+  if (currPlayer == 1) {
+    player1Num = playerNum;
+  } else {
+    player2Num = playerNum;
+  }
+};
+
+/**
+ * Automate choosing the player's number from the dice rolls
+ * Return the player number
+ * @param {number} firstNumeralIndex
+ */
```

Comment:
@jeremylim91 what's this? Let's not leave dangling comments- this is confusing and not professional